### PR TITLE
End-to-end Actor-Critic Training

### PR DESCRIPTION
**Pull request by Chrisytz**
To reproduce the 50.6 accuracy obtained from running VERL on TPU please use this command
```
python3 -m verl.trainer.main_ppo data.train_files=$HOME/data/gsm8k/train.parquet data.val_files=$HOME/data/gsm8k/test.parquet data.train_batch_size=128 data.max_prompt_length=256 data.max_response_length=256 actor_rollout_ref.model.path=Qwen/Qwen2.5-0.5B-Instruct actor_rollout_ref.actor.optim.lr=1e-6 actor_rollout_ref.actor.ppo_mini_batch_size=32 actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=2 actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=8 actor_rollout_ref.rollout.tensor_model_parallel_size=1 actor_rollout_ref.rollout.gpu_memory_utilization=0.2 actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=4 critic.optim.lr=1e-5 critic.model.path=Qwen/Qwen2.5-0.5B-Instruct critic.ppo_micro_batch_size_per_gpu=2 algorithm.kl_ctrl.kl_coef=0.001 trainer.logger=['console'] trainer.val_before_train=False trainer.default_hdfs_dir=null trainer.n_gpus_per_node=1 trainer.nnodes=1 trainer.save_freq=10 trainer.test_freq=10 trainer.total_epochs=15 trainer.n_tpus_per_node=1 actor_rollout_ref.rollout.enable_sleep_mode=False actor_rollout_ref.rollout.enforce_eager=False actor_rollout_ref.rollout.free_cache_engine=False actor_rollout_ref.rollout.disable_mm_preprocessor_cache=False data.truncation=right 2>&1 | tee verl_demo.log
```

* initializing the vllm in the actor worker's `_build_rollout` method, leads to `requires_grad=False` and `grad_fn = None` during actor and critic updates, so I wrapped the update in a `torch.enable_grad()` context manager (not quite sure why)

### CODE REVIEW

**Comment by bhavya01** [comment]
Waiting for https://github.com/Chrisytz/verl/pull/8 to merge first

**Comment by Chrisytz** [comment]
I'm going to clean up some unused code first, I'll let you guys know when it's ready for review!

File: verl/workers/actor/dp_actor.py [hunk 1]
```diff
@@ -42,7 +42,11 @@
     from flash_attn.bert_padding import index_first_axis, pad_input, rearrange, unpad_input
 elif is_npu_available:
     from transformers.integrations.npu_flash_attention import index_first_axis, pad_input, rearrange, unpad_input
-
+else:
+    try:
+        import torch_xla
+    except:
```
Comment (qihqi) [inline verl/workers/actor/dp_actor.py:48]: except ImportError, ModuleNotFound

**Review by qihqi — APPROVED** [review]
looks good!
